codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/SymbolVisitor.kt

```diff
@@ -63,14 +63,16 @@ data class SymbolVisitorConfig(
     val runtimeConfig: RuntimeConfig,
     val codegenConfig: CodegenConfig,
     val handleOptionality: Boolean = true,
```

Comment:
let's fully delete `handleOptionality` as an option and just have it be what we do when `handleRequired` is false

Reply:
currently there's a bit of a weird case where if you set `handleOptionality` to false right now, we also won't handle required

Reply:
Yep, you are right, it can go away. I kept it for compatibility only.